### PR TITLE
Fixes #20239 - Add upgrade script for qpid path

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

File: hooks/pre/30-upgrade.rb [hunk 1]
```diff
@@ -102,6 +102,36 @@ def mark_qpid_cert_for_update
   end
 end
 
+# rubocop:disable MethodLength
+# rubocop:disable Style/MultipleComparison
+def upgrade_qpid_paths
+  qpid_dir = '/var/lib/qpidd'
+  qpid_data_dir = "#{qpid_dir}/.qpidd"
```
Comment (pmoravec) [inline hooks/pre/30-upgrade.rb]: This works well on RHEL7. For RHEL6:

    qpid_data_dir = "#{qpid_dir}"

Comment (chris1984) [inline hooks/pre/30-upgrade.rb]: Going to address this with a downstream MR since upstream is only supporting EL7

File: hooks/pre/30-upgrade.rb [hunk 2]
```diff
@@ -102,6 +102,36 @@ def mark_qpid_cert_for_update
...
+    Kafo::Helpers.execute("mkdir -p #{qpid_linearstore}/p001/efp/2048k/returned")
+    Kafo::Helpers.execute("mkdir -p #{qpid_linearstore}/jrnl2")
+    # Copy dat dir to new location dat2
+    Kafo::Helpers.execute("cp -r #{qpid_linearstore}/dat #{qpid_linearstore}/dat2")
+    # Move qpid jrnl files
+    Dir.foreach("#{qpid_linearstore}/jrnl") do |item|
+      next if item == '.' || item == '..'
+      Kafo::Helpers.execute("puts Moving #{item}")
+      Kafo::Helpers.execute("mv #{qpid_dir}/#{item} #{qpid_linearstore}/p001/efp/2048k/in_use/#{item}")
+      Kafo::Helpers.execute("ln -s #{qpid_linearstore}/p001/efp/2048k/in_use/#{item} #{qpid_linearstore}/jrnl2/#{item}")
```
Comment (pmoravec) [inline hooks/pre/30-upgrade.rb]: These two lines need to be performed in a loop over the #{qpid_linearstore}/jrnl/#{item} directory - for every file in this directory, move it to `in_use` dir, and create a symlink to it. So something like:

    Dir.foreach("#{qpid_linearstore}/jrnl/#{item}") do |jrnlfile|
        Kafo::Helpers.execute("mv #{qpid_linearstore}/jrnl/#{item}/#{jrnlfile} #{qpid_linearstore}/p001/efp/2048k/in_use/#{jrnlfile}")
        Kafo::Helpers.execute("ln -s #{qpid_linearstore}/p001/efp/2048k/in_use/#{jrnlfile} #{qpid_linearstore}/jrnl2/#{item}/#{jrnlfile}")
    end

File: hooks/pre/30-upgrade.rb [hunk 3]
```diff
@@ -157,6 +187,7 @@ def fail_and_exit(message)
     upgrade_step :start_httpd, :run_always => true
     upgrade_step :start_qpidd, :run_always => true
     upgrade_step :start_pulp, :run_always => true
+    upgrade_step :upgrade_qpid_paths
```
Comment (pmoravec) [inline hooks/pre/30-upgrade.rb]: I dont know how the upgrade steps are ordered, but the `upgrade_qpid_paths` must be run when `qpidd` is stopped (so when it is recommended to have stopped also `pulp` or `candlepin` and so).

Comment (chris1984) [inline hooks/pre/30-upgrade.rb]: Ah ok, will change that around to have it done before anything is running

File: hooks/pre/30-upgrade.rb [hunk 4]
```diff
@@ -102,6 +102,36 @@ def mark_qpid_cert_for_update
...
+    # Make new directory structure for migration
+    Kafo::Helpers.execute("mkdir -p #{qpid_linearstore}/p001/efp/2048k/in_use")
+    Kafo::Helpers.execute("mkdir -p #{qpid_linearstore}/p001/efp/2048k/returned")
+    Kafo::Helpers.execute("mkdir -p #{qpid_linearstore}/jrnl2")
+    # Copy dat dir to new location dat2
+    Kafo::Helpers.execute("cp -r #{qpid_linearstore}/dat #{qpid_linearstore}/dat2")
+    # Move qpid jrnl files
+    Dir.foreach("#{qpid_linearstore}/jrnl") do |item|
+      next if item == '.' || item == '..'
+      Kafo::Helpers.execute("puts Moving #{item}")
```
Comment (ehelms) [inline hooks/pre/30-upgrade.rb]: I think you want the puts to just be a line and not wrapped in `execute`

**Comment by ehelms** [comment]
What happens if any step fails? Should we consider a rescue to catch errors and spit out a more friendly error followed by any backtrace associated to a failure?

**Comment by chris1984** [comment]
@pmoravec @ehelms can I get another review please when you get a chance.

**Comment by pmoravec** [comment]
Looks good.

**Pull request merged**